### PR TITLE
[MM-853]: Added server testcases for store/subscription_store.go

### PR DESCRIPTION
**Pull request by Kshitij-Katiyar**
### Summary
Added server testcases for store/subscription_store.go

### CODE REVIEW

File: calendar/store/subscription_store_test.go [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+
+	"github.com/mattermost/mattermost-plugin-mscalendar/calendar/remote"
+
+	"github.com/mattermost/mattermost/server/public/model"
+)
+
+func TestLoadSubscription(t *testing.T) {
+	mockAPI, store, _, _, _ := GetMockSetup(t)
```
Comment (raghavaggarwal2308) [inline calendar/store/subscription_store_test.go]: move this inside

File: calendar/store/subscription_store_test.go [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+
+	tests := []struct {
+		name       string
+		setup      func()
+		assertions func(*testing.T, *Subscription, error)
+	}{
+		{
+			name: "Error loading subscription",
+			setup: func() {
+				mockAPI.On("KVGet", mock.Anything).Return(nil, &model.AppError{Message: "Subscription not found"}).Times(1)
```
Comment (raghavaggarwal2308) [inline calendar/store/subscription_store_test.go]: pass proper value

File: calendar/store/subscription_store_test.go [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+			assertions: func(t *testing.T, sub *Subscription, err error) {
+				require.Error(t, err)
+				require.Nil(t, sub)
+				require.EqualError(t, err, "failed plugin KVGet: Subscription not found")
+			},
+		},
+		{
+			name: "Successful Load",
+			setup: func() {
+				mockAPI.On("KVGet", mock.Anything).Return([]byte(`{"PluginVersion":"1.0","Remote":{"ID":"mockRemoteID","CreatorID":"mockCreatorID"}}`), nil).Times(1)
```
Comment (raghavaggarwal2308) [inline calendar/store/subscription_store_test.go]: Create util for response value

File: calendar/store/subscription_store_test.go [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+
+			tt.assertions(t, err)
+
+			mockAPI.AssertExpectations(t)
+		})
+	}
+}
+
+func TestDeleteUserSubscription(t *testing.T) {
+	mockAPI, store, mockLogger, mockLoggerWith, _ := GetMockSetup(t)
```
Comment (raghavaggarwal2308) [inline calendar/store/subscription_store_test.go]: move inside

File: calendar/store/subscription_store_test.go [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+	mockUser := &User{MattermostUserID: "user1", Settings: Settings{EventSubscriptionID: "mockEventSubscriptionID"}, Remote: &remote.User{ID: "mockRemoteUserID"}}
```
Comment (raghavaggarwal2308) [inline calendar/store/subscription_store_test.go]: I think we created a util for this?

**Review by raghavaggarwal2308 — CHANGES_REQUESTED** [review]

File: calendar/store/subscription_store_test.go (hunk 3 above)
Comment (Kshitij-Katiyar) [reply calendar/store/subscription_store_test.go]: used only once

File: calendar/store/subscription_store_test.go (hunk 2 above)
Comment (Kshitij-Katiyar) [reply calendar/store/subscription_store_test.go:26]: generated at run time with random values

**Review by raghavaggarwal2308 — APPROVED** [review]

**Pull request merged**